src/engine/Screen.ts

```diff
@@ -0,0 +1,500 @@
+   * String must be in the format of css style background-position. The vertical position must precede the horizontal position in strings.
+   *
+   * Valid String examples: "top left", "top", "bottom", "middle", "middle center", "bottom right"
+   * Valid [[AbsolutePosition]] examples: `{top: 5, right: 10%}`, `{bottom: 49em, left: 10px}`, `{left: 10, bottom: 40}`
+   */
+  position?: CanvasPosition;
+}
+
+export class Screen {
+  private _canvas: HTMLCanvasElement;
+  private _browser: BrowserEvents;
+  private _camera: Camera;
+  private _resolution: ScreenDimension;
+  private _resolutionStack: ScreenDimension[] = [];
+  private _viewport: ScreenDimension;
+  private _viewportStack: ScreenDimension[] = [];
+  private _pixelRatio: number | null = null;
+  private _position: CanvasPosition;
+  private _displayMode: DisplayMode;
+  private _isFullScreen = false;
+  private _logger = Logger.getInstance();
+  constructor(options: ScreenOptions) {
+    this.resolution = options.resolution;
+    this.viewport = options.viewport ?? this.resolution;
+    this._displayMode = options.displayMode ?? DisplayMode.Fixed;
+    this._canvas = options.canvas;
+    this._browser = options.browser;
+    this._position = options.position;
+    this._pixelRatio = options.pixelRatio;
+    this._applyDisplayMode();
+    this._onPixelRatioChange('pixelRatioChange', () => {
+      this._logger.debug('Pixel Ratio Change', window.devicePixelRatio);
+      this.applyResolutionAndViewport();
+    });
+  }
+
+  public get pixelRatio(): number {
+    if (this._pixelRatio) {
+      return this._pixelRatio;
+    }
+
+    if (window.devicePixelRatio < 1) {
+      return 1;
+    }
+
+    const devicePixelRatio = window.devicePixelRatio || 1;
+
+    return devicePixelRatio;
+  }
+
+  public get isHiDpi() {
+    return this.pixelRatio !== 1;
+  }
+
+  public get displayMode(): DisplayMode {
+    return this._displayMode;
+  }
+
+  public get canvas(): HTMLCanvasElement {
+    return this._canvas;
+  }
+
+  public get resolution(): ScreenDimension {
+    return this._resolution;
+  }
+
+  public set resolution(resolution: ScreenDimension) {
+    this._resolution = resolution;
+  }
+
+  public get viewport(): ScreenDimension {
+    if (this._viewport) {
+      return this._viewport;
+    }
+    return this._resolution;
+  }
+
+  public set viewport(viewport: ScreenDimension) {
+    this._viewport = viewport;
+  }
+
+  public get scaledWidth() {
+    return this._resolution.width * this.pixelRatio;
+  }
+
+  public get scaledHeight() {
+    return this._resolution.height * this.pixelRatio;
+  }
+
+  public setCurrentCamera(camera: Camera) {
+    this._camera = camera;
+  }
+
+  private _onPixelRatioChange(_event: 'pixelRatioChange', handler: () => any) {
+    const mqString = `(resolution: ${window.devicePixelRatio}dppx)`;
+    matchMedia(mqString).addEventListener('change', handler);
```

Comment:
We should consider cleaning up event listeners. Each time a new engine is created, we'd be adding these listeners, right?

Also should this be:

```suggestion
    this._browser.window.matchMedia(mqString).addEventListener('change', handler);
```